src/Arcus.WebApi.Unit/Security/Authentication/SharedAccessKeyAuthenticationController.cs

```diff
@@ -10,8 +10,8 @@ public class SharedAccessKeyAuthenticationController : ControllerBase
     {
         [HttpGet]
         [Route("authz/shared-access-key")]
-        [SharedAccessKeyAuthentication(headerName: "x-shared-access-key", secretName: "custom-access-key-name")]
-        public Task<IActionResult> TestHardCodedConfiguredSharedAccessKeyHeaderName(HttpRequestMessage message)
+        [SharedAccessKeyAuthentication(headerName: "x-shared-access-key", queryParameterName: "api-key", secretName: "custom-access-key-name")]
+        public Task<IActionResult> TestHardCodedConfiguredSharedAccessKey(HttpRequestMessage message)
```

Comment:
I might be exaggerating but I would target this endpoint and an isolated one per auth mechanism approach.

With that, I mean that every query parameter-based tests should call this one, but also a new one which only has queryParameterName configured and not `headerName` to ensure that both scenarios are covered, if I'm making myself clear?

Reply:
understood, will look into this, but it also relates to a previous comment regarding both header and querystring parameter are present at same time

Reply:
See other conversation

Reply:
@tomkerkhove This has been updated in latest PR